src/lib/vm/vm.js

```diff
@@ -1085,7 +1085,49 @@ class VmStack {
           });
 
           return undefined;
-        } else if (callee === "setTimeout") {
+        } else if (callee === "useMemo" || callee === "useCallback") {
+          if (this.prevStack) {
+              throw new Error(
+                  `Method: ${callee}. The hook can only be called from the top of the stack`
+              );
+          }
+      
+          const isMemo = callee === "useMemo";
+          const fnArgName = isMemo ? 'factory' : 'callback';
+          if (args.length < 1) {
+              throw new Error(
+                  `Method: ${callee}. Required arguments: '${fnArgName}'. Optional: 'dependencies'`
+              );
+          }
+      
+          const fn = args[0];
+          if (!(fn instanceof Function)) {
+              throw new Error(
+                  `Method: ${callee}. The first argument '${fnArgName}' must be a function`
+              );
+          }
+      
+          const hookIndex = this.hookIndex++;
```

Comment:
Please check that `this.vm.hooks` exists, similar to `useState`. Otherwise it would be an issue with `VM.require`